config/configuration.toml.in

```diff
@@ -20,6 +20,17 @@ default_vcpus = -1
 #default_memory = @DEFMEMSZ@
 disable_block_device_use = @DEFDISABLEBLOCK@
 
+# Enable pre allocation of VM RAM, default false
+#enable_mem_prealloc = true
+#
+# Enable Realtime QEMU
+#enable_realtime = true
+#
+# Enable Mlock 
```

Comment:
Can you expand the title further. Could this be `Enable memory locking` for example?

Also, `qemu(1)` only seems to provide a single option that handles both realtime and mlock, so why do we need (and how are we providing) two knobs?

Reply:
By default memory once allocated is always locked. Unfortunately to be able to swap memory in qemu the realtime option must be enabled. 

So our default behavior is locked non pre-allocated. So memory is allocated on behalf of the VM, when the memory is actually touched by the VM. Once memory is brought in, it is not swapped out.

So our behaviour is

1. Default, non pre-allocated but locked (with none of the knobs set)
2. Realtime + Locked
    Which gives you realtime behaviour
3. Realtime + Unlocked (i.e. swap enabled)
    Here what we desire is swap enabled, but we have to enable realtime which is unfortunate.
    But once you enable swap, it is not realtime anyway. So effectively it is just enabling swap.
    So in the implementation, we can choose to honor the mlock=off w/o requiring the realtime 
    knob to be set.
   
4. You can also enable pre-allocated, realtime and locked (which does not work today btw. The agent seems to be failing. 
   Note: This is not an issue for normal VM's. It just triggers some error in Clear Containers, the agent in particular.

There will be bug filed for 4, once this PR is merged to track that issue.